### DISCUSSION TITLE
DSC PC 1808 Unrecognized data

### DISCUSSION THREAD

**Discussion by Tiaanh** [taligentx/dscKeybusInterface#38] [dscKeybusInterface Discussion]
Hallo,

I've tested on a PC 1808 but get "Unrecognized data". Any pointers where I should start to debug ?

  298.46: 00000000 0 [0x00] Unrecognized data[CRC OK]
  298.69: 00000000 0 00000000 00000000 00000000 00000000 [0x00] Unrecognized data[CRC OK]
  298.73: 00000000 [0x00] Unrecognized data[CRC OK]
  298.74: 00000000 0 [0x00] Unrecognized data[CRC OK]
  298.76: 00000000 0 00000000 [0x00] Unrecognized data[CRC OK]
  298.80: 00000000 0 [0x00] Unrecognized data[CRC OK]
  298.88: 00000000 0 [0x00] Unrecognized data[CRC OK]
  298.96: 00000000 0 [0x00] Unrecognized data[CRC OK]
  299.04: 00000000 0 [0x00] Unrecognized data[CRC OK]

Thanks

**Comment by taligentx** [comment]
Hi @Tiaanh, 

Which microcontroller are you using, and how is it connected?

**Comment by Tiaanh** [comment]
I'm using an Arduino Uno. It's connected as per the Wiring section of the notes, I don't have the "Viirtual keypad" connected and the Uno is powered from USB.

Thanks
Tiaan

**Comment by Tiaanh** [comment]
I have two jumper cables linked to extent the length. The green jumper from the Uno connect to the black cable and in turn connected to the "DSC Green" connection. The yellow jumper from the Uno connect to the white cable and in turn connect to the "DSC Yellow" connection. The purple cable on the DSC Aux(-) connect to the Unu's ground via the bread board.

![20180824_184548-1612x1209](https://user-images.githubusercontent.com/28781573/44597826-3c82ab00-a7d1-11e8-838e-5a9c6aed1301.jpg)

![20180824_184654-1612x1209](https://user-images.githubusercontent.com/28781573/44597852-545a2f00-a7d1-11e8-872f-707d267c06a6.jpg)

**Comment by l33d** [comment]
@Tiaanh

Hi, 

I had same problem. Bad connection from alarm or arduino. Please follow the schematic diagram and use properly GPIO on esp or arduino in sketch.

**Comment by ColinNg** [comment]
One of Nikhil's suggestions - the WeMos D1 mini - is amazing. Here's mine, with a Power shield so I can take power straight from AUX+/AUX-. I unsoldered the barrel jack and added in Resistors and used a 2N7000 FET (it does the same as the NPN transistor - but it uses 1 less resistor). Total cost is < $5 *shipped*. 

![img_0131](https://user-images.githubusercontent.com/137680/44615372-e3754000-a7ed-11e8-9afe-5b0ce77c94ef.jpg)

**Comment by taligentx** [comment]
>I have two jumper cables linked to extent the length. The green jumper from the Uno connect to the black cable and in turn connected to the "DSC Green" connection. The yellow jumper from the Uno connect to the white cable and in turn connect to the "DSC Yellow" connection. The purple cable on the DSC Aux(-) connect to the Unu's ground via the bread board.

@Tiaanh I haven't tested the interface using jumpers and breadboards, it's possible that the multiple connection interfaces are causing an issue.  I recommend directly soldering the wiring and resistors, it's few enough components that I don't even use a perfboard and simply solder the components inline.

**Comment by taligentx** [comment]
@ColinNg Nice work!  Adding a note to the todo list to document the power options in README.md.

**Comment by Tiaanh** [comment]
Soldering and connecting without breadboard solved the issue. Thank you very much for the help.
